### PR TITLE
Add PDDL

### PR DESCRIPTION
**Pull request by celrm**
## Description

Adds support for [PDDL](https://planning.wiki/guide/whatis/pddl).

## Checklist:

- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      -  https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3Apddl+NOT+nothack
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - https://github.com/MylerRL661/AI-Pathfinding
      - https://github.com/EL2425/Drone_Project_2017-
    - Sample license(s):
      - https://github.com/MylerRL661/AI-Pathfinding/blob/main/LICENSE
      - https://github.com/EL2425/Drone_Project_2017-/blob/master/LICENSE.md
  - [x] I have included a syntax highlighting grammar: https://github.com/jan-dolejsi/vscode-pddl

### CODE REVIEW

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -4622,6 +4622,14 @@ P4:
   tm_scope: source.p4
   ace_mode: text
   language_id: 348895984
+PDDL:
+  type: programming
+  color: "#0d00ff"
+  extensions:
+  - ".pddl"
+  tm_scope: none
```
Comment (lildude) [inline lib/linguist/languages.yml:4630]: You're going to need a scope for syntax highlighting. This will be the scope in the grammar.

**Review by lildude — CHANGES_REQUESTED** [review]
See inline comment.

We also don't need quite so many nearly identical samples, not the large ones. Please remove the samples that are collapsed in the diff (clear indicator they're too big) and pick the most diverse two or three of the remaining samples and remove the others.

File: lib/linguist/languages.yml (hunk 1 above)
Comment (lildude) [reply lib/linguist/languages.yml:4630]:
```suggestion
  tm_scope: source.pddl
```

**Comment by celrm** [comment]
> See inline comment.
> 
> We also don't need quite so many nearly identical samples, not the large ones. Please remove the samples that are collapsed in the diff (clear indicator they're too big) and pick the most diverse two or three of the remaining samples and remove the others.

Thank you, I have applied the suggestions now. Please let me know if there is anything else.

**Comment by lildude** [comment]
Running `script/list-grammars` and committing the changes should fix the tests.

**Review by lildude — APPROVED** [review]

**Pull request merged**